internal/services/notificationhub/notification_hub_namespace_resource.go

```diff
@@ -117,18 +132,29 @@ func resourceNotificationHubNamespaceCreate(d *pluginsdk.ResourceData, meta inte
 		return tf.ImportAsExistsError("azurerm_notification_hub_namespace", id.ID())
 	}
 
+	zoneRedundancy := namespaces.ZoneRedundancyPreferenceDisabled
+	if v, ok := d.GetOk("zone_redundancy_enabled"); ok && v.(bool) {
+		zoneRedundancy = namespaces.ZoneRedundancyPreferenceEnabled
+	}
+
 	namespaceType := namespaces.NamespaceType(d.Get("namespace_type").(string))
 	parameters := namespaces.NamespaceResource{
 		Location: location.Normalize(d.Get("location").(string)),
 		Sku: namespaces.Sku{
 			Name: namespaces.SkuName(d.Get("sku_name").(string)),
 		},
 		Properties: &namespaces.NamespaceProperties{
-			NamespaceType: &namespaceType,
-			Enabled:       utils.Bool(d.Get("enabled").(bool)),
+			NamespaceType:  &namespaceType,
+			Enabled:        pointer.To(d.Get("enabled").(bool)),
+			ZoneRedundancy: pointer.To(zoneRedundancy),
 		},
 		Tags: tags.Expand(d.Get("tags").(map[string]interface{})),
 	}
+
+	if v, ok := d.GetOk("replication_region"); ok {
+		parameters.Properties.ReplicationRegion = pointer.To(namespaces.ReplicationRegion(azure.NormalizeLocation(v.(string))))
```

Comment:
`azure.NormalizeLocation` is deprecated and points to `location.Normalize` 

```suggestion
		parameters.Properties.ReplicationRegion = pointer.To(namespaces.ReplicationRegion(location.Normalize(v.(string))))
```